ruby/ql/lib/codeql/files/FilesImpl.qll

```diff
@@ -0,0 +1,21 @@
+private import codeql.utils.FileSystem as FS
```

Comment:
This should probably be in a subfolder named `internal` , there is no reason external users should ever import this module. This is also why it was defined as a `private module` initially.

Reply:
A problem with that is that I want to use `module FilesImpl` elsewhere later, the module will not just be used for instantiating the shared File implementation.  
So it feels wrong to put `FilesImpl` in a `internal/` folder. 

E.g. my shared ReDoS implementation will depend on `FS::FilesSig` (or something similar), so I will need the `FilesImpl` module when I instantiate the parameterised ReDoS implementation.   
And it feels wrong to import `codeql.files.internal.FilesImpl` all the way from `codeql.ruby.security.regexp.ExponentialBackTracking`.  

In the glorious future we might have many packs/modules from the `shared/` folder that depend on each other, so this issue will only become bigger. 

Another possible solution would be to create a `ruby/ql/lib/codeql/ruby/utils` folder, and that folder would be dedicated to implementations of signatures from the `utils` package.   
(And the other languages would have a similar folder).  
I kinda like that solution.  
An alternative folder name would be `impls/` or `impl/`, and that folder name could also be used elsewhere (e.g. I might create an `impls/` folder under `codeql.ruby.security.regexp`).  

Other implementations will hopefully be easier to place. E.g. the instantiations of the ReDoS library will just be in the same files that the current implementation sits in.

Reply:
I think it is OK for _us_ to use `internal` libraries wherever we want, we just like to make sure that nobody else assumes stability of the internal ibraries.

Reply:
I still think external developers should be able to use e.g. `FileSystemImpl` if they develop their own shared library.  

However, it's easy to later move something out of an `internal/` folder, but it's tricky to move some into an `internal/` folder without breaking our deprecation policy.  
So I'll move the implementations into an `internal/` folder.